compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2789,6 +2777,10 @@ private BType checkIndexAccessExpr(BLangIndexBasedAccess indexBasedAccessExpr, B
                     actualType = ((BArrayType) varRefType).getElementType();
                 }
                 break;
+            case TypeTags.TUPLE:
+                indexExprType = checkExpr(indexExpr, this.env, symTable.noType);
```

Comment:
This will set the type to the `indexExpr`. Then we don't have to pass the `indexExprType` to the `checkTupleIndexBasedAccess()` below since we can access the type. WDYT?